posix/stdio.c

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/internal/posix/device.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/trace.h>
+#include "mount.h"
+
+int oe_rename(const char* oldpath, const char* newpath)
+{
+    int ret = -1;
+    oe_device_t* fs = NULL;
+    oe_device_t* newfs = NULL;
+    char filepath[OE_PATH_MAX];
```

Comment:
Consider allocating the 8KB for these two buffers on the heap instead.

Reply:
Fixed. Using this pattern:

```
int oe_rename(const char* oldpath, const char* newpath)
{
    int ret = -1;
    oe_device_t* fs = NULL;
    oe_device_t* newfs = NULL;
    typedef struct _variables
    {
        char filepath[OE_PATH_MAX];
        char newfilepath[OE_PATH_MAX];
    }
    variables_t;
    variables_t* v = NULL;

    if (!(v = oe_malloc(sizeof(variables_t))))
        OE_RAISE_ERRNO(OE_ENOMEM);

    /* Use v->filepath and v->newfilepath here */

 done:

    if (v)
        oe_free(v);

    return ret;
}
```